environments/common/inventory/group_vars/all/openondemand.yml

```
@@ -12,11 +12,12 @@ openondemand_servername: "{{ hostvars[groups['openondemand'].0].ansible_host if
 
 openondemand_auth: basic_pam
 
-openondemand_jupyter_partition: "{{ openhpc_partitions[0]['name'] }}"
-openondemand_desktop_partition: "{{ openhpc_partitions[0]['name'] }}"
-openondemand_rstudio_partition: "{{ openhpc_partitions[0]['name'] }}"
+# Manages case where openhpc_partitions are not defined e.g. for site image builds
+openondemand_jupyter_partition: "{{ openhpc_partitions[0].name if (openhpc_partitions | default([])) else '' }}"
```

So what this has missed is that in https://github.com/stackhpc/ansible-slurm-appliance/issues/680#issuecomment-3460798861 we said:
> - define cluster_compute_groups in common envs, empty, so that without any nodes deployed/inventory file we get empty openhpc_nodegroups and empty openhpc_partitions.


So with that done, I think this is/can just be the more obvious/ideomatic:

```suggestion
openondemand_jupyter_partition: "{{ openhpc_partitions[0].name | default('') }}"
```

---

And, given that "truthy" is not always obvious, is it better to make the default `false`?